pkgs/applications/window-managers/eww/default.nix

```diff
@@ -10,16 +10,18 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "eww";
-  version = "0.2.0";
+  version = "0.3.0";
 
   src = fetchFromGitHub {
     owner = "elkowar";
     repo = pname;
     rev = "v${version}";
-    sha256 = "050zc3w1z9f2vg6sz86mdxf345gd3s3jf09gf4y8y1mqkzs86b8x";
+    sha256 = "055il2b3k8x6mrrjin6vkajpksc40phcp4j1iq0pi8v3j7zsfk1a";
   };
 
-  cargoSha256 = "sha256-LejnTVv9rhL9CVW1fgj2gFv4amHQeziu5uaH2ae8AAw=";
+  cargoSha256 = "sha256-3hGA730g8E4rwQ9V0wSLUcAEmockXi+spwp50cgf0Mw=";
+
+  cargoPatches = [ ./Cargo.lock.patch ];
```

Comment:
I would suggest to use substituteInPlace if possible.

Reply:
https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/rust.section.md#building-a-crate-with-an-absent-or-out-of-date-cargolock-file-building-a-crate-with-an-absent-or-out-of-date-cargolock-file

This suggests using `cargoPatches`

Reply:
Usually when you'd patch a lockfile it's because the dependencies haven't been updated or the file doesn't exist at all.

For replacing one version string substituteInPlace will do fine.

Reply:
I do not see `substituteInPlace` here, as the match is too complicated. We want to avoid substituting the wrong 0.2.0 in the process!

At the same time, we also want the build to fail in the moment that the patch doesn't apply anymore.

In my opinion it is either a patchfile to be used here, or some mechanism that understands the TOML and can emit an updated TOML, similar to how one would/could use `jq` do patch a JSON file.

Reply:
> In my opinion it is either a patchfile to be used here, or some mechanism that understands the TOML and can emit an updated TOML, similar to how one would/could use `jq` do patch a JSON file.

Thats over engineered. We don't do that.